src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -43,11 +43,14 @@ namespace System.Text.RegularExpressions
         private List<string>? _capnamelist;
 
         private RegexOptions _options;
-        private ValueListBuilder<RegexOptions> _optionsStack;
+        // NOTE: _optionsStack is ValueListBuilder<int> to ensure that
+        //       ArrayPool<int>.Shared, not ArrayPool<RegexOptions>.Shared,
```

Comment:
Do we have other places in the tree where renting from an `ArrayPool<int>` would be better than any other, rarely used, `ArrayPool<T>`?

Reply:
Some git-grepping later, and found no direct uses of `ArrayPool<T>` where T isn't a primitive type, and other uses of `ValueListBuilder<T>` seems to only use `T=int` with the exception of this place here.